src/sa/sa.c

```diff
@@ -86,12 +93,31 @@ int sa_pton(const char *addr, struct sa *sa)
 			sa->u.in6.sin6_family = AF_INET6;
 		}
 	}
+	else if (!strncmp(addr, "fe80:", 5) && strchr(addr, '%')) {
```

Comment:
I think the previous if statement will be triggered first.
inet_pton on linux or osx handles % without reading the value

Reply:
`inet_pton()` is preferred. If it does not work, `getaddrinfo()` is tried. Should I change something?